### PR TITLE
[Bugfix] Correctly check when to set/unset "full screen" mode

### PR DESCRIPTION
**Pull request by josokinas**
# Problem

1) There might be an edge case when previewing an image in full screen and changing the step - "full screen" mode is not disabled.
2) Currently decision to switch on/off "full screen" mode relies on internal state rather than on global one.

# Solution

1) When component responsible for making an image to "cover the whole container" is unmounted and "full screen" mode is on - it's then turned off.
2) An action to switch "full screen" on/off is now determining based on global state.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [n/a] Have new manual tests been written down?
- [x] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by josokinas** [comment]
This fixes https://github.com/onfido/onfido-sdk-ui/issues/578.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ This project adheres to the Node [default version scheme](https://docs.npmjs.com
 
 ### Fixed
 - Public: Handle non JSON error responses and return a `Connection Lost` error to the user
+- Public: Make sure "full screen" mode is exited when navigating away from enlarged preview
```
Comment (rfreitas) [inline CHANGELOG.md]: it's actually UI, since it's doesn't interfere with the sdk api (that's the case that makes it "public")

Comment (josokinas) [reply CHANGELOG.md]: Oh, thanks, will change that.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-641-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://641-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**